onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc

```diff
@@ -203,65 +214,76 @@ bool HasValidQuantizationZeroPoints(const InitializedTensorSet& initializers, co
-      int64_t zero_dim = zero_tensor.dims().empty() ? 1 : zero_tensor.dims()[0];
-      if (is_conv_u8s8_weight) {
-        if (zero_tensor.data_type() != ONNX_NAMESPACE::TensorProto_DataType_INT8) {
-          LOGS_DEFAULT(VERBOSE) << "u8s8 QlinearConv only supports int8 zero point for weight, "
-                                << "actual zero point type: [" << zero_tensor.data_type() << "]";
+      if (zero_dim != 1) {
+        if (is_qlinear_matmul) {
+          LOGS_DEFAULT(VERBOSE) << "QLinearMatMul does not support per-channel quantization";
           return false;
         }
+      }
 
-        // For onnx, u8s8 QlinearConv, the weight zero point can be a scalar,
-        // or a tensor with same channel as weight, for NNAPI we only support it be
-        // 0 (scalar) or all 0 (tensor), NNAPI will assume the zero point for per-channel
-        // quantization is 0 there is no input for it
-        const auto& weight_tensor = *initializers.at(node.InputDefs()[3]->Name());
-        if (weight_tensor.dims()[0] != zero_dim && zero_dim != 1) {
-          LOGS_DEFAULT(VERBOSE) << op_type << " mismatch int8 per-channel quantization weight,"
-                                << " weight dimension[0] " << weight_tensor.dims()[0]
-                                << " zero point dimension " << zero_dim;
-          return false;
-        }
+      // For onnx, u8s8 QlinearConv, the weight zero point can be a scalar,
+      // or a tensor with same channel as weight, for NNAPI we only support it be
+      // 0 (scalar) or all 0 (tensor), NNAPI will assume the zero point for per-channel
+      // quantization is 0 there is no input for it
+      const auto& weight_tensor = *initializers.at(node.InputDefs()[3]->Name());
+      if (weight_tensor.dims()[0] != zero_dim && zero_dim != 1) {
+        LOGS_DEFAULT(VERBOSE) << op_type << " mismatch int8 per-channel quantization weight,"
+                              << " weight dimension[0] " << weight_tensor.dims()[0]
+                              << " zero point dimension " << zero_dim;
+        return false;
+      }
 
-        std::unique_ptr<uint8_t[]> unpacked_tensor;
-        size_t tensor_byte_size;
-        auto status = onnxruntime::utils::UnpackInitializerData(
-            zero_tensor,
-            node.ModelPath(),
-            unpacked_tensor, tensor_byte_size);
-        if (!status.IsOK()) {
-          LOGS_DEFAULT(ERROR) << "QLinearConv erro when unpack zero tensor:" << status.ErrorMessage();
-          return false;
-        }
+      std::unique_ptr<uint8_t[]> unpacked_tensor;
+      size_t tensor_byte_size;
+      auto status = onnxruntime::utils::UnpackInitializerData(
+          zero_tensor,
+          node.ModelPath(),
+          unpacked_tensor, tensor_byte_size);
+      if (!status.IsOK()) {
+        LOGS_DEFAULT(ERROR) << "Qlinear[Conv/MatMul] error when unpack zero tensor:" << status.ErrorMessage();
+        return false;
```

Comment:
nit: unpacking

Would be good to include the initializer name if it's not in the Status error message so it's more easily debugged.